src/sentry/incidents/subscription_processor.py

```diff
@@ -0,0 +1,198 @@
+from __future__ import absolute_import
+
+import logging
+import operator
+from datetime import timedelta
+
+from django.conf import settings
+from django.db import transaction
+
+from sentry.incidents.logic import (
+    alert_aggregation_to_snuba,
+    create_incident,
+    update_incident_status,
+)
+from sentry.incidents.models import (
+    AlertRule,
+    AlertRuleAggregations,
+    AlertRuleThresholdType,
+    Incident,
+    IncidentStatus,
+    IncidentType,
+)
+from sentry.utils import metrics, redis
+from sentry.utils.dates import to_datetime
+
+
+logger = logging.getLogger(__name__)
+
+
+class SubscriptionProcessor(object):
+    """
+    Class for processing subscription updates for an alert rule. Accepts a subscription
+    and then can process one or more updates via `process_update`. Keeps track of how
```

Comment:
I wrote this so that it could potentially accept multiple updates. In practice we won't do this, and probably shouldn't since the db state could potentially change.